lib/sharing/editor_ui.dart

```diff
@@ -0,0 +1,86 @@
+import 'dart:async';
+
+import 'package:logging/logging.dart';
+
+import '../context.dart';
+import '../dart_pad.dart';
+import '../editing/editor.dart';
+import '../elements/analysis_results_controller.dart';
+import '../elements/elements.dart';
+import '../services/common.dart';
+import '../services/dartservices.dart';
+
+abstract class EditorUi {
+  final Logger logger = Logger('dartpad');
+  ContextBase get context;
+
+  Future analysisRequest;
```

Comment:
I think this can be `Future<void>`, or something like that, rather than `Future` (of dynamic); whatever `dartServices.analyze` returns.